gammapy/scripts/image_pipe.py

```diff
@@ -64,9 +66,34 @@ def make_counts(self, obs_id):
         obs = self.data_store.obs(obs_id=obs_id)
         events = obs.events.select_energy(self.energy_band)
         events = events.select_offset(self.offset_band)
-        counts_map.fill(events=events)
-        counts_map.data = counts_map.data.value
-        self.maps["counts"] = counts_map
+        if (len(events) != 0):
+            counts_map.fill(events=events)
+            counts_map.data = counts_map.data.value
+            self.maps["counts"] = counts_map
+        else:
+            self.maps["counts"] = counts_map
+
+    def _has_counts(self, obs_id):
```

Comment:
I take it back.

You have a background modeling technique that doesn't work at all if you have no counts.
So you have to skip runs with zero counts completely, i.e. not accumulate background and exposure.
Maybe emit a warning?

We'll have to think more about how to handle different background modeling methods with the same image / cube analysis pipeline...